pkg/volume/plugins.go

```diff
@@ -743,7 +743,11 @@ func (pm *VolumePluginMgr) refreshProbedPlugins() {
 	events, err := pm.prober.Probe()
 	if err != nil {
 		klog.Errorf("Error dynamically probing plugins: %s", err)
-		return // Use cached plugins upon failure.
+		//skipping events will lead to fail to add all health plugins driver once one plugin file is not ready, always keeping consume health events.
+		if events == nil {
```

Comment:
I don't believe this block is actually doing anything - if events is nil, the for loop will just not iterate and the end result is the same.

I think the right solution is to update the implementation of `Probe()` to return separate errors for each volume plugin, and reuse the cache only for plugins with errors. The current solution isn't safe given that some events could be undefined when iterated upon.

/cc @linyouchong

Reply:
`Probe()` could be changed to return a struct with each entry like
```
struct{
  event
  error err
}
```
 so we can get an error per event thus increasing granularity and being able to work on the events that did not have errors.

Just an idea, I don't actually know what probe does or what the response of it should look like